ci/azure-pipelines-e2e.yml

```diff
@@ -0,0 +1,187 @@
+  displayName: 'Check project dependencies'
+ 
+- script: |
+    echo "Installing minikube..."
+    curl -Lo minikube https://storage.googleapis.com/minikube/releases/$(MinikubeVersion)/minikube-linux-amd64 && chmod +x minikube && sudo mv minikube /usr/local/bin/
+    echo "Installing kubeadm..."
+    curl -Lo kubeadm https://storage.googleapis.com/kubernetes-release/release/$(KubernetesVersion)/bin/linux/amd64/kubeadm && chmod +x kubeadm && sudo mv kubeadm /usr/bin/
+    echo "Installing kubectl..."     
+    curl -Lo kubectl https://storage.googleapis.com/kubernetes-release/release/$(KubernetesVersion)/bin/linux/amd64/kubectl && chmod +x kubectl && sudo mv kubectl /usr/local/bin/
+  displayName: 'Prepare clusterctl dependencies'
+
+- script: |
+    sudo -E minikube start --vm-driver=none --bootstrapper=localkube --kubernetes-version=$(KubernetesVersion) --feature-gates=CustomResourceSubresources=true --extra-config=apiserver.Authorization.Mode=RBAC
+    sudo chown -R $USER $HOME/.kube
+    sudo chgrp -R $USER $HOME/.kube
+    sudo chown -R $USER $HOME/.minikube
+    sudo chgrp -R $USER $HOME/.minikube
+    # Update kube config
+    minikube update-context
+    
+    # Update certs
+    sed -i "s/certificate-authority.*/certificate-authority-data: $(cat $HOME/.minikube/ca.crt | base64 -w0)/" $HOME/.kube/config
+    sed -i "s/client-certificate.*/client-certificate-data: $(cat $HOME/.minikube/client.crt | base64 -w0)/" $HOME/.kube/config
+    sed -i "s/client-key.*/client-key-data: $(cat $HOME/.minikube/client.key | base64 -w0)/" $HOME/.kube/config
+  displayName: 'Start and configure minikube'
+
+- script: |
+    # Copy config to etc/kubernetes
+    sudo mkdir -p /etc/kubernetes
+    sudo cp $HOME/.kube/config /etc/kubernetes/admin.conf
+    sudo chown $USER:$USER /etc/kubernetes/admin.conf    
+    # wait for nodes be ready and update node label
+    JSONPATH='{range .items[*]}{@.metadata.name}:{range @.status.conditions[*]}{@.type}={@.status};{end}{end}'; until kubectl get nodes -o jsonpath="$JSONPATH" 2>&1 | grep -q "Ready=True"; do sleep 1; echo "waiting for node to become ready"; kubectl get nodes; done
+    NODE=$(kubectl get nodes -o jsonpath="{range .items[*]}{@.metadata.name}{end}")
+    kubectl label node ${NODE} node-role.kubernetes.io/master=
+    # Give cluster-admin permissions to the default service account
+    kubectl create clusterrolebinding add-on-cluster-admin --clusterrole=cluster-admin --serviceaccount=kube-system:default
+    # Wait for kube-dns to be ready
+    JSONPATH='{range .items[*]}{@.metadata.name}:{range @.status.conditions[*]}{@.type}={@.status};{end}{end}'; until kubectl -n kube-system get pods -lk8s-app=kube-dns -o jsonpath="$JSONPATH" 2>&1 | grep -q "Ready=True"; do sleep 1;echo "waiting for kube-dns to be available"; kubectl get pods --all-namespaces; done    
+    echo "Validating kubectl configuration..."
+    kubectl cluster-info
+  displayName: 'Configure kubernetes'
+
+- script: |
```

Comment:
locally builds the Docker images. This is readily available to minikube with the `vm-driver=none` option.